packages/ubus-lime-location/files/usr/libexec/rpcd/lime-location

```diff
@@ -81,19 +71,17 @@ local function nodes_and_links()
         table.insert(links, string.lower(mac))
     end
   end
-      result = {}
-      result[hostname] = {
-      hostname=hostname,
-      macs=macs,
-      coordinates=coordinates,
-      links=links
-      }
-	printJson(result)
+  result = {}
+  result[hostname] = {hostname=hostname, macs=macs, coordinates=coordinates, links=links}
+  return result
+end
 
+local function nodes_and_links()
+  utils.printJson(_nodes_and_links())
 end
 
 function all_nodes_and_links(msg)
-  print('{"result":'..shell("shared-state get nodes_and_links")..'}')
+  print('{"result":'..utils.unsafe_shell("shared-state get nodes_and_links")..'}')
```

Comment:
Eventually shared-state could be called a a library instead of as a shell. Not for now of course...

Reply:
Yes, I thought about this a little. We may export an easier/ergonomic shared-state API as now to use it some stuff must be done (locks have to be taken, released, etc).